core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,194 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation Jobs for exploration models"""
+
+from __future__ import annotations
+
+from urllib.parse import urlparse   # pylint: disable-msg=C0003
+
+from core.domain import exp_fetchers
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import bs4
+from typing import Iterator
+
+(exp_models, ) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class GetExpsWithInvalidURLJob(base_jobs.JobBase):
+    """Job that returns invalid exploration models."""
+
+    # The following function has been adapted from
+    # https://stackoverflow.com/questions/39233973/get-all-keys-of-a-nested-dictionary
+    def recursive_items(self, dictionary: dict) -> Iterator[tuple]:
+        """Yields an iterator containing tuples of key, value pairs
+
+        Yields:
+            tuple. Yields tuples of key, value pairs.
+        """
+        for key, value in dictionary.items():
+            if isinstance(value, dict):
+                yield from self.recursive_items(value)
+            else:
+                yield (key, value)
+
+    def get_invalid_links(self, dictionary: dict) -> list:
+        """Returns list of invalid links
+
+            Returns:
+                list. Returns list of invalid links.
+            """
+        string = ''
+        for key, value in self.recursive_items(dictionary):
+            if key == 'html':
+                string += value
+
+        soup = bs4.BeautifulSoup(string, 'html.parser')
+
+        # Extract links.
+        links = soup.find_all('oppia-noninteractive-link')
+        cleaned_links = []
+        for link in links:
+            # Remove &quot; from the links.
+            cleaned_links.append(
+                link.get('url-with-value').replace('&quot;', ''))
+
+        invalid_links = []
+        for link in cleaned_links:
+            # Separate protocol from the link.
```

Comment:
This comment is not really applicable rn.

Reply:
Removed this comment.